crates/pipeline_manager/src/auth.rs

```diff
@@ -0,0 +1,368 @@
+use std::{collections::HashMap, env};
```

Comment:
maybe add module level comments using //! that explains the high-level concepts would be helpful